networking/src/NTP/Client.hs

```diff
@@ -45,26 +46,29 @@ import           NTP.Packet (NtpPacket (..), evalClockOffset, mkCliNtpPacket, nt
 import           NTP.Util (createAndBindSock, resolveNtpHost, selectIPv4, selectIPv6,
                            udpLocalAddresses, withSocketsDoLifted)
 
-data NtpClientSettings m = NtpClientSettings
-    { ntpServers         :: [String]
-      -- ^ list of servers addresses
-    , ntpHandler         :: (Microsecond, Microsecond) -> m ()
-      -- ^ got time callback (margin, time when client sent request)
-    , ntpLogName         :: LoggerName
-      -- ^ logger name modifier
-    , ntpResponseTimeout :: Microsecond
-      -- ^ delay between making requests and response collection
-    , ntpPollDelay       :: Microsecond
-      -- ^ how often to send responses to server
-    , ntpMeanSelection   :: [(Microsecond, Microsecond)] -> (Microsecond, Microsecond)
-      -- ^ way to sumarize results received from different servers.
-      -- this may accept list of lesser size than @length ntpServers@ in case some servers
-      -- failed to respond in time, but never an empty list
-    }
+data NtpClientSettings m =
```

Comment:
While we're here, let's make `NtpClientSettings` monomorphic in `IO`. I don't anticipate any difficulties here, and we can get rid of the hoisting functions, the `NtpMonad` constraint, use of `MonadMask`, etc.